frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/storage/UploadImageModel.java

```diff
@@ -371,12 +372,18 @@ public boolean flush() {
             DiskImage diskImage = (DiskImage) getDiskModel().getDisk();
             diskImage.setSize(getVirtualSize());
             diskImage.setActualSizeInBytes(imageInfoModel.getActualSize());
-            diskImage.setVolumeFormat(diskModel.getIsIncrementalBackup().getEntity() ?
-                    VolumeFormat.COW : getImageInfoModel().getFormat());
-            diskImage.setVolumeType(AsyncDataProvider.getInstance().getVolumeType(
-                    diskImage.getVolumeFormat(),
-                    getDiskModel().getStorageDomain().getSelectedItem().getStorageType(), null, null));
-            diskImage.setContentType(getImageInfoModel().getContentType());
+            VolumeFormat volumeFormat = diskModel.getIsIncrementalBackup().getEntity() ?
+                    VolumeFormat.COW : getImageInfoModel().getFormat();
+            diskImage.setVolumeFormat(volumeFormat);
+            DiskContentType contentType = getImageInfoModel().getContentType();
+            if (volumeFormat != VolumeFormat.COW && contentType == DiskContentType.ISO) {
```

Comment:
I think we can just add the additional if:
```
if (diskImage.getVolumeFormat() != VolumeFormat.COW && getImageInfoModel().getContentType() == DiskContentType.ISO) {
                diskImage.setVolumeType(VolumeType.Preallocated);
            }
```
And skip the other changes :)

Reply:
@dupondje Thanks for the suggestion.
I applied the changes (with additional comments).
Could you please review once more?
Thanks in advance.